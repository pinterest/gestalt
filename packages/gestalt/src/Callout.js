// @flow strict
import React, { type Node } from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';
import Box from './Box.js';
import Heading from './Heading.js';
import Icon from './Icon.js';
import IconButton from './IconButton.js';
import Button from './Button.js';
import Text from './Text.js';
import { useColorScheme } from './contexts/ColorScheme.js';
import styles from './Callout.css';
import useResponsiveMinWidth from './useResponsiveMinWidth.js';
import { type ActionDataType, type DismissButtonType } from './commonTypes.js';

type Props = {|
  dismissButton?: DismissButtonType,
  iconAccessibilityLabel: string,
  message: string,
  primaryAction?: ActionDataType,
  secondaryAction?: ActionDataType,
  type: 'error' | 'info' | 'warning',
  title?: string,
|};

const CALLOUT_TYPE_ATTRIBUTES = {
  info: {
    icon: 'info-circle',
    color: 'blue',
    backgroundColor: '#EBF4FE',
  },
  warning: {
    icon: 'workflow-status-warning',
    color: 'orange',
    backgroundColor: '#FDF5EC',
  },
  error: {
    icon: 'workflow-status-problem',
    color: 'red',
    backgroundColor: '#FDEBEE',
  },
};

const CalloutAction = ({
  data,
  stacked,
  type,
}: {|
  data: ActionDataType,
  stacked?: boolean,
  type: string,
|}): Node => {
  const { name: colorSchemeName } = useColorScheme();
  const isDarkMode = colorSchemeName === 'darkMode';
  let color = type === 'primary' ? 'white' : 'transparent';
  if (isDarkMode && type === 'secondary') {
    color = 'transparentWhiteText';
  }
  const { accessibilityLabel, label, onClick, href } = data;

  return (
    <Box
      display="block"
      smDisplay="flex"
      alignItems="center"
      justifyContent="center"
      paddingX={1}
      marginTop={type === 'secondary' && stacked ? 2 : undefined}
      smMarginTop="auto"
      smMarginBottom="auto"
    >
      {href ? (
        <Button
          accessibilityLabel={accessibilityLabel}
          color={color}
          href={href}
          onClick={onClick}
          role="link"
          size="lg"
          text={label}
        />
      ) : (
        <Button
          accessibilityLabel={accessibilityLabel}
          color={color}
          onClick={onClick}
          role="button"
          size="lg"
          text={label}
        />
      )}
    </Box>
  );
};

export default function Callout({
  dismissButton,
  iconAccessibilityLabel,
  message,
  primaryAction,
  secondaryAction,
  type,
  title,
}: Props): Node {
  // Currently there is not a dark mode spec for this component. This is to ensure
  // that all text is readable.
  const { name } = useColorScheme();
  const isDarkMode = name === 'darkMode';
  const responsiveMinWidth = useResponsiveMinWidth();

  return (
    <Box
      dangerouslySetInlineStyle={{
        __style: {
          backgroundColor: CALLOUT_TYPE_ATTRIBUTES[type].backgroundColor,
        },
      }}
      display="flex"
      direction="column"
      smDirection="row"
      padding={6}
      smPadding={8}
      position="relative"
      rounding={4}
    >
      <Box smDisplay="flex" wrap width="100%" smMarginTop={-3} smMarginBottom={-3}>
        <Box
          display="flex"
          direction="column"
          smDirection="row"
          justifyContent="center"
          alignItems="center"
          marginBottom={primaryAction || secondaryAction ? 4 : undefined}
          smMarginBottom={primaryAction || secondaryAction ? 0 : undefined}
          smPaddingY={3}
        >
          <Box marginBottom={4} marginTop={0} smMarginBottom="auto" smMarginTop="auto">
            <Icon
              accessibilityLabel={iconAccessibilityLabel}
              color={CALLOUT_TYPE_ATTRIBUTES[type].color}
              icon={CALLOUT_TYPE_ATTRIBUTES[type].icon}
              size={32}
            />
          </Box>
          <Box marginBottom="auto" marginTop="auto" maxWidth={648} paddingX={6}>
            <Box
              display="flex"
              smDisplay="block"
              direction="column"
              alignItems="center"
              marginBottom="auto"
              marginTop="auto"
            >
              {title && (
                <Box marginBottom={2}>
                  <Heading align={responsiveMinWidth === 'xs' ? 'center' : undefined} size="sm">
                    {title}
                  </Heading>
                </Box>
              )}
              <Text align={responsiveMinWidth === 'xs' ? 'center' : undefined}>{message}</Text>
            </Box>
          </Box>
        </Box>
        {(primaryAction || secondaryAction) && (
          <Box smDisplay="flex" marginStart="auto" smMarginEnd={4} smPaddingY={3}>
            {secondaryAction && responsiveMinWidth !== 'xs' && (
              <CalloutAction type="secondary" data={secondaryAction} />
            )}
            {primaryAction && <CalloutAction type="primary" data={primaryAction} />}
            {secondaryAction && responsiveMinWidth === 'xs' && (
              <CalloutAction type="secondary" data={secondaryAction} stacked={!!secondaryAction} />
            )}
          </Box>
        )}
      </Box>

      {dismissButton && (
        <div className={classnames(styles.rtlPos)}>
          <IconButton
            accessibilityLabel={dismissButton.accessibilityLabel}
            icon="cancel"
            iconColor={isDarkMode ? 'white' : 'darkGray'}
            onClick={dismissButton.onDismiss}
            padding={4}
            size="sm"
          />
        </div>
      )}
    </Box>
  );
}

Callout.propTypes = {
  // $FlowFixMe[signature-verification-failure] flow 0.135.0 upgrade
  dismissButton: PropTypes.exact({
    accessibilityLabel: PropTypes.string.isRequired,
    onDismiss: PropTypes.func.isRequired,
  }),
  iconAccessibilityLabel: PropTypes.string.isRequired,
  message: PropTypes.string.isRequired,
  // $FlowFixMe[signature-verification-failure] flow 0.135.0 upgrade
  primaryAction: PropTypes.shape({
    href: PropTypes.string,
    label: PropTypes.string.isRequired,
    onClick: PropTypes.func,
    accessibilityLabel: PropTypes.string,
  }),
  // $FlowFixMe[signature-verification-failure] flow 0.135.0 upgrade
  secondaryAction: PropTypes.shape({
    href: PropTypes.string,
    label: PropTypes.string.isRequired,
    onClick: PropTypes.func,
    accessibilityLabel: PropTypes.string,
  }),
  // $FlowFixMe[signature-verification-failure] flow 0.135.0 upgrade
  type: PropTypes.oneOf(['error', 'info', 'warning']).isRequired,
  title: PropTypes.string,
};
