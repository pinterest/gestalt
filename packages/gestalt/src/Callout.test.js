// @flow strict
import { create } from 'react-test-renderer';
import Callout from './Callout.js';
import Text from './Text.js';

describe('<Callout />', () => {
  test('Error Callout', () => {
    const tree = create(
      <Callout
        message="Insert a clever error callout message here"
        iconAccessibilityLabel="error"
        type="error"
      />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Warning Callout', () => {
    const tree = create(
      <Callout
        message="Insert a clever warning callout message here"
        iconAccessibilityLabel="warning"
        type="warning"
      />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Info Callout', () => {
    const tree = create(
      <Callout
        message="Insert a clever info callout message here"
        iconAccessibilityLabel="info"
        type="info"
      />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('message + title', () => {
    const tree = create(
      <Callout
        message="Insert a clever info callout message here"
        iconAccessibilityLabel="info"
        type="info"
        title="A Title"
      />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('message + title + primaryAction with href', () => {
    const tree = create(
      <Callout
        message="Insert a clever info callout message here"
        iconAccessibilityLabel="info"
        primaryAction={{ href: 'pinterest.com', label: 'Visit Pinterest', accessibilityLabel: '' }}
        type="info"
        title="A Title"
      />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('message + title + primaryAction without href', () => {
    const tree = create(
      <Callout
        message="Insert a clever info callout message here"
        iconAccessibilityLabel="info"
        primaryAction={{ label: 'Visit Pinterest', accessibilityLabel: '' }}
        type="info"
        title="A Title"
      />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('message + title + primaryAction + secondaryAction', () => {
    const tree = create(
      <Callout
        message="Insert a clever info callout message here"
        iconAccessibilityLabel="info"
        primaryAction={{ href: 'pinterest.com', label: 'Visit Pinterest', accessibilityLabel: '' }}
        secondaryAction={{
          href: 'pinterest.com/help',
          label: 'Learn more',
          accessibilityLabel: 'Learn more: callout messages',
        }}
        type="info"
        title="A Title"
      />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('message + title + primaryAction + dismissButton', () => {
    const tree = create(
      <Callout
        message="Insert a clever info callout message here"
        iconAccessibilityLabel="info"
        primaryAction={{ href: 'pinterest.com', label: 'Visit Pinterest', accessibilityLabel: '' }}
        dismissButton={{
          accessibilityLabel: 'Dismiss banner',
          onDismiss: () => {},
        }}
        type="info"
        title="A Title"
      />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('callout with rich text message', () => {
    const tree = create(
      <Callout
        dismissButton={{
          accessibilityLabel: 'Dismiss this banner',
          onDismiss: () => {},
        }}
        iconAccessibilityLabel="Info"
        message={
          <Text inline>
            You have invited{' '}
            <Text inline weight="bold">
              Leaf Media Agency
            </Text>{' '}
            to your business hierarchy. Once they accept, you will be able to manage their business
            account.
          </Text>
        }
        primaryAction={{
          accessibilityLabel: 'Resend invite',
          label: 'Resend invite',
        }}
        secondaryAction={{
          accessibilityLabel: 'Cancel invite',
          label: 'Cancel invite',
        }}
        title="You've sent an invite"
        type="info"
      />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
