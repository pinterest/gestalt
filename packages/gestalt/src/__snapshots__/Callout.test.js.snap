// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Callout /> Error Callout 1`] = `
<div
  className="box errorWeak paddingX6 paddingY6 relative rounding4 smDirectionRow smPaddingX8 smPaddingY8 xsDirectionColumn xsDisplayFlex"
>
  <div
    className="box flexWrap smDisplayFlex smMarginBottomN3 smMarginTopN3"
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <div
      className="box justifyCenter smDirectionRow smPaddingY3 xsDirectionColumn xsDisplayFlex xsItemsCenter"
    >
      <div
        className="box marginBottom4 marginTop0 smMarginBottomAuto smMarginTopAuto"
      >
        <svg
          aria-hidden={null}
          aria-label="error"
          className="icon errorIcon iconBlock"
          height={32}
          role="img"
          viewBox="0 0 24 24"
          width={32}
        >
          <path
            d="test-file-stub"
          />
        </svg>
      </div>
      <div
        className="box marginBottomAuto marginTopAuto paddingX6"
        style={
          Object {
            "maxWidth": 648,
          }
        }
      >
        <div
          className="box marginBottomAuto marginTopAuto smDisplayBlock xsDirectionColumn xsDisplayFlex xsItemsCenter"
        >
          <div
            className="Text fontSize300 defaultText alignStart breakWord fontWeightNormal"
          >
            Insert a clever error callout message here
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Callout /> Info Callout 1`] = `
<div
  className="box infoWeak paddingX6 paddingY6 relative rounding4 smDirectionRow smPaddingX8 smPaddingY8 xsDirectionColumn xsDisplayFlex"
>
  <div
    className="box flexWrap smDisplayFlex smMarginBottomN3 smMarginTopN3"
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <div
      className="box justifyCenter smDirectionRow smPaddingY3 xsDirectionColumn xsDisplayFlex xsItemsCenter"
    >
      <div
        className="box marginBottom4 marginTop0 smMarginBottomAuto smMarginTopAuto"
      >
        <svg
          aria-hidden={null}
          aria-label="info"
          className="icon infoIcon iconBlock"
          height={32}
          role="img"
          viewBox="0 0 24 24"
          width={32}
        >
          <path
            d="test-file-stub"
          />
        </svg>
      </div>
      <div
        className="box marginBottomAuto marginTopAuto paddingX6"
        style={
          Object {
            "maxWidth": 648,
          }
        }
      >
        <div
          className="box marginBottomAuto marginTopAuto smDisplayBlock xsDirectionColumn xsDisplayFlex xsItemsCenter"
        >
          <div
            className="Text fontSize300 defaultText alignStart breakWord fontWeightNormal"
          >
            Insert a clever info callout message here
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Callout /> Warning Callout 1`] = `
<div
  className="box paddingX6 paddingY6 relative rounding4 smDirectionRow smPaddingX8 smPaddingY8 warningWeak xsDirectionColumn xsDisplayFlex"
>
  <div
    className="box flexWrap smDisplayFlex smMarginBottomN3 smMarginTopN3"
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <div
      className="box justifyCenter smDirectionRow smPaddingY3 xsDirectionColumn xsDisplayFlex xsItemsCenter"
    >
      <div
        className="box marginBottom4 marginTop0 smMarginBottomAuto smMarginTopAuto"
      >
        <svg
          aria-hidden={null}
          aria-label="warning"
          className="icon warningIcon iconBlock"
          height={32}
          role="img"
          viewBox="0 0 24 24"
          width={32}
        >
          <path
            d="test-file-stub"
          />
        </svg>
      </div>
      <div
        className="box marginBottomAuto marginTopAuto paddingX6"
        style={
          Object {
            "maxWidth": 648,
          }
        }
      >
        <div
          className="box marginBottomAuto marginTopAuto smDisplayBlock xsDirectionColumn xsDisplayFlex xsItemsCenter"
        >
          <div
            className="Text fontSize300 defaultText alignStart breakWord fontWeightNormal"
          >
            Insert a clever warning callout message here
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Callout /> callout with rich text message 1`] = `
<div
  className="box infoWeak paddingX6 paddingY6 relative rounding4 smDirectionRow smPaddingX8 smPaddingY8 xsDirectionColumn xsDisplayFlex"
>
  <div
    className="box flexWrap smDisplayFlex smMarginBottomN3 smMarginTopN3"
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <div
      className="box justifyCenter marginBottom4 smDirectionRow smMarginBottom0 smPaddingY3 xsDirectionColumn xsDisplayFlex xsItemsCenter"
    >
      <div
        className="box marginBottom4 marginTop0 smMarginBottomAuto smMarginTopAuto"
      >
        <svg
          aria-hidden={null}
          aria-label="Info"
          className="icon infoIcon iconBlock"
          height={32}
          role="img"
          viewBox="0 0 24 24"
          width={32}
        >
          <path
            d="test-file-stub"
          />
        </svg>
      </div>
      <div
        className="box marginBottomAuto marginTopAuto paddingX6"
        style={
          Object {
            "maxWidth": 648,
          }
        }
      >
        <div
          className="box marginBottomAuto marginTopAuto smDisplayBlock xsDirectionColumn xsDisplayFlex xsItemsCenter"
        >
          <div
            className="box marginBottom2"
          >
            <div
              className="Text fontSize400 defaultText alignStart breakWord fontWeightSemiBold"
            >
              You've sent an invite
            </div>
          </div>
          <span
            className="Text fontSize300 defaultText alignStart breakWord fontWeightNormal"
          >
            You have invited
             
            <span
              className="Text fontSize300 defaultText alignStart breakWord fontWeightSemiBold"
            >
              Leaf Media Agency
            </span>
             
            to your business hierarchy. Once they accept, you will be able to manage their business account.
          </span>
        </div>
      </div>
    </div>
    <div
      className="box marginStartAuto smDisplayFlex smMarginEnd4 smPaddingY3"
    >
      <div
        className="box justifyCenter paddingX1 smDisplayFlex smMarginBottomAuto smMarginTopAuto xsDisplayBlock xsItemsCenter"
      >
        <button
          aria-label="Cancel invite"
          className="button hideOutline block accessibilityOutline parentButton"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onMouseDown={[Function]}
          onMouseUp={[Function]}
          onTouchCancel={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <div
            className="button hideOutline block accessibilityOutline tapTransition lg transparent enabled childrenDiv"
          >
            <div
              className="Text fontSize300 defaultText alignCenter fontWeightSemiBold"
            >
              Cancel invite
            </div>
          </div>
        </button>
      </div>
      <div
        className="box justifyCenter paddingX1 smDisplayFlex smMarginBottomAuto smMarginTopAuto xsDisplayBlock xsItemsCenter"
      >
        <button
          aria-label="Resend invite"
          className="button hideOutline block accessibilityOutline parentButton"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onMouseDown={[Function]}
          onMouseUp={[Function]}
          onTouchCancel={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <div
            className="button hideOutline block accessibilityOutline tapTransition lg white enabled childrenDiv"
          >
            <div
              className="Text fontSize300 defaultText alignCenter fontWeightSemiBold"
            >
              Resend invite
            </div>
          </div>
        </button>
      </div>
    </div>
  </div>
  <div
    className="rtlPos"
  >
    <button
      aria-label="Dismiss this banner"
      className="parentButton"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchCancel={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <div
        className="button tapTransition enabled"
      >
        <div
          className="pog transparent"
          style={
            Object {
              "height": 48,
              "width": 48,
            }
          }
        >
          <svg
            aria-hidden={true}
            aria-label=""
            className="icon defaultIcon iconBlock"
            height={16}
            role="img"
            viewBox="0 0 24 24"
            width={16}
          >
            <path
              d="test-file-stub"
            />
          </svg>
        </div>
      </div>
    </button>
  </div>
</div>
`;

exports[`<Callout /> message + title + primaryAction + dismissButton 1`] = `
<div
  className="box infoWeak paddingX6 paddingY6 relative rounding4 smDirectionRow smPaddingX8 smPaddingY8 xsDirectionColumn xsDisplayFlex"
>
  <div
    className="box flexWrap smDisplayFlex smMarginBottomN3 smMarginTopN3"
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <div
      className="box justifyCenter marginBottom4 smDirectionRow smMarginBottom0 smPaddingY3 xsDirectionColumn xsDisplayFlex xsItemsCenter"
    >
      <div
        className="box marginBottom4 marginTop0 smMarginBottomAuto smMarginTopAuto"
      >
        <svg
          aria-hidden={null}
          aria-label="info"
          className="icon infoIcon iconBlock"
          height={32}
          role="img"
          viewBox="0 0 24 24"
          width={32}
        >
          <path
            d="test-file-stub"
          />
        </svg>
      </div>
      <div
        className="box marginBottomAuto marginTopAuto paddingX6"
        style={
          Object {
            "maxWidth": 648,
          }
        }
      >
        <div
          className="box marginBottomAuto marginTopAuto smDisplayBlock xsDirectionColumn xsDisplayFlex xsItemsCenter"
        >
          <div
            className="box marginBottom2"
          >
            <div
              className="Text fontSize400 defaultText alignStart breakWord fontWeightSemiBold"
            >
              A Title
            </div>
          </div>
          <div
            className="Text fontSize300 defaultText alignStart breakWord fontWeightNormal"
          >
            Insert a clever info callout message here
          </div>
        </div>
      </div>
    </div>
    <div
      className="box marginStartAuto smDisplayFlex smMarginEnd4 smPaddingY3"
    >
      <div
        className="box justifyCenter paddingX1 smDisplayFlex smMarginBottomAuto smMarginTopAuto xsDisplayBlock xsItemsCenter"
      >
        <a
          aria-label=""
          className="link noUnderline hideOutline tapTransition pill accessibilityOutline flex justifyCenter xsItemsCenter button lg white"
          href="pinterest.com"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyPress={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchCancel={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          rel=""
          tabIndex={0}
          target={null}
        >
          <div
            className="Flex rowGap2 columnGap0 justifyCenter xsDirectionRow xsItemsCenter"
          >
            <div
              className="FlexItem"
            >
              <div
                className="Text fontSize300 defaultText alignCenter fontWeightSemiBold"
              >
                Visit Pinterest
              </div>
            </div>
          </div>
        </a>
      </div>
    </div>
  </div>
  <div
    className="rtlPos"
  >
    <button
      aria-label="Dismiss banner"
      className="parentButton"
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onMouseDown={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchCancel={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <div
        className="button tapTransition enabled"
      >
        <div
          className="pog transparent"
          style={
            Object {
              "height": 48,
              "width": 48,
            }
          }
        >
          <svg
            aria-hidden={true}
            aria-label=""
            className="icon defaultIcon iconBlock"
            height={16}
            role="img"
            viewBox="0 0 24 24"
            width={16}
          >
            <path
              d="test-file-stub"
            />
          </svg>
        </div>
      </div>
    </button>
  </div>
</div>
`;

exports[`<Callout /> message + title + primaryAction + secondaryAction 1`] = `
<div
  className="box infoWeak paddingX6 paddingY6 relative rounding4 smDirectionRow smPaddingX8 smPaddingY8 xsDirectionColumn xsDisplayFlex"
>
  <div
    className="box flexWrap smDisplayFlex smMarginBottomN3 smMarginTopN3"
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <div
      className="box justifyCenter marginBottom4 smDirectionRow smMarginBottom0 smPaddingY3 xsDirectionColumn xsDisplayFlex xsItemsCenter"
    >
      <div
        className="box marginBottom4 marginTop0 smMarginBottomAuto smMarginTopAuto"
      >
        <svg
          aria-hidden={null}
          aria-label="info"
          className="icon infoIcon iconBlock"
          height={32}
          role="img"
          viewBox="0 0 24 24"
          width={32}
        >
          <path
            d="test-file-stub"
          />
        </svg>
      </div>
      <div
        className="box marginBottomAuto marginTopAuto paddingX6"
        style={
          Object {
            "maxWidth": 648,
          }
        }
      >
        <div
          className="box marginBottomAuto marginTopAuto smDisplayBlock xsDirectionColumn xsDisplayFlex xsItemsCenter"
        >
          <div
            className="box marginBottom2"
          >
            <div
              className="Text fontSize400 defaultText alignStart breakWord fontWeightSemiBold"
            >
              A Title
            </div>
          </div>
          <div
            className="Text fontSize300 defaultText alignStart breakWord fontWeightNormal"
          >
            Insert a clever info callout message here
          </div>
        </div>
      </div>
    </div>
    <div
      className="box marginStartAuto smDisplayFlex smMarginEnd4 smPaddingY3"
    >
      <div
        className="box justifyCenter paddingX1 smDisplayFlex smMarginBottomAuto smMarginTopAuto xsDisplayBlock xsItemsCenter"
      >
        <a
          aria-label="Learn more: callout messages"
          className="link noUnderline hideOutline tapTransition pill accessibilityOutline flex justifyCenter xsItemsCenter button lg transparent"
          href="pinterest.com/help"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyPress={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchCancel={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          rel=""
          tabIndex={0}
          target={null}
        >
          <div
            className="Flex rowGap2 columnGap0 justifyCenter xsDirectionRow xsItemsCenter"
          >
            <div
              className="FlexItem"
            >
              <div
                className="Text fontSize300 defaultText alignCenter fontWeightSemiBold"
              >
                Learn more
              </div>
            </div>
          </div>
        </a>
      </div>
      <div
        className="box justifyCenter paddingX1 smDisplayFlex smMarginBottomAuto smMarginTopAuto xsDisplayBlock xsItemsCenter"
      >
        <a
          aria-label=""
          className="link noUnderline hideOutline tapTransition pill accessibilityOutline flex justifyCenter xsItemsCenter button lg white"
          href="pinterest.com"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyPress={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchCancel={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          rel=""
          tabIndex={0}
          target={null}
        >
          <div
            className="Flex rowGap2 columnGap0 justifyCenter xsDirectionRow xsItemsCenter"
          >
            <div
              className="FlexItem"
            >
              <div
                className="Text fontSize300 defaultText alignCenter fontWeightSemiBold"
              >
                Visit Pinterest
              </div>
            </div>
          </div>
        </a>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Callout /> message + title + primaryAction with href 1`] = `
<div
  className="box infoWeak paddingX6 paddingY6 relative rounding4 smDirectionRow smPaddingX8 smPaddingY8 xsDirectionColumn xsDisplayFlex"
>
  <div
    className="box flexWrap smDisplayFlex smMarginBottomN3 smMarginTopN3"
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <div
      className="box justifyCenter marginBottom4 smDirectionRow smMarginBottom0 smPaddingY3 xsDirectionColumn xsDisplayFlex xsItemsCenter"
    >
      <div
        className="box marginBottom4 marginTop0 smMarginBottomAuto smMarginTopAuto"
      >
        <svg
          aria-hidden={null}
          aria-label="info"
          className="icon infoIcon iconBlock"
          height={32}
          role="img"
          viewBox="0 0 24 24"
          width={32}
        >
          <path
            d="test-file-stub"
          />
        </svg>
      </div>
      <div
        className="box marginBottomAuto marginTopAuto paddingX6"
        style={
          Object {
            "maxWidth": 648,
          }
        }
      >
        <div
          className="box marginBottomAuto marginTopAuto smDisplayBlock xsDirectionColumn xsDisplayFlex xsItemsCenter"
        >
          <div
            className="box marginBottom2"
          >
            <div
              className="Text fontSize400 defaultText alignStart breakWord fontWeightSemiBold"
            >
              A Title
            </div>
          </div>
          <div
            className="Text fontSize300 defaultText alignStart breakWord fontWeightNormal"
          >
            Insert a clever info callout message here
          </div>
        </div>
      </div>
    </div>
    <div
      className="box marginStartAuto smDisplayFlex smMarginEnd4 smPaddingY3"
    >
      <div
        className="box justifyCenter paddingX1 smDisplayFlex smMarginBottomAuto smMarginTopAuto xsDisplayBlock xsItemsCenter"
      >
        <a
          aria-label=""
          className="link noUnderline hideOutline tapTransition pill accessibilityOutline flex justifyCenter xsItemsCenter button lg white"
          href="pinterest.com"
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyPress={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchCancel={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          rel=""
          tabIndex={0}
          target={null}
        >
          <div
            className="Flex rowGap2 columnGap0 justifyCenter xsDirectionRow xsItemsCenter"
          >
            <div
              className="FlexItem"
            >
              <div
                className="Text fontSize300 defaultText alignCenter fontWeightSemiBold"
              >
                Visit Pinterest
              </div>
            </div>
          </div>
        </a>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Callout /> message + title + primaryAction without href 1`] = `
<div
  className="box infoWeak paddingX6 paddingY6 relative rounding4 smDirectionRow smPaddingX8 smPaddingY8 xsDirectionColumn xsDisplayFlex"
>
  <div
    className="box flexWrap smDisplayFlex smMarginBottomN3 smMarginTopN3"
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <div
      className="box justifyCenter marginBottom4 smDirectionRow smMarginBottom0 smPaddingY3 xsDirectionColumn xsDisplayFlex xsItemsCenter"
    >
      <div
        className="box marginBottom4 marginTop0 smMarginBottomAuto smMarginTopAuto"
      >
        <svg
          aria-hidden={null}
          aria-label="info"
          className="icon infoIcon iconBlock"
          height={32}
          role="img"
          viewBox="0 0 24 24"
          width={32}
        >
          <path
            d="test-file-stub"
          />
        </svg>
      </div>
      <div
        className="box marginBottomAuto marginTopAuto paddingX6"
        style={
          Object {
            "maxWidth": 648,
          }
        }
      >
        <div
          className="box marginBottomAuto marginTopAuto smDisplayBlock xsDirectionColumn xsDisplayFlex xsItemsCenter"
        >
          <div
            className="box marginBottom2"
          >
            <div
              className="Text fontSize400 defaultText alignStart breakWord fontWeightSemiBold"
            >
              A Title
            </div>
          </div>
          <div
            className="Text fontSize300 defaultText alignStart breakWord fontWeightNormal"
          >
            Insert a clever info callout message here
          </div>
        </div>
      </div>
    </div>
    <div
      className="box marginStartAuto smDisplayFlex smMarginEnd4 smPaddingY3"
    >
      <div
        className="box justifyCenter paddingX1 smDisplayFlex smMarginBottomAuto smMarginTopAuto xsDisplayBlock xsItemsCenter"
      >
        <button
          aria-label=""
          className="button hideOutline block accessibilityOutline parentButton"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onMouseDown={[Function]}
          onMouseUp={[Function]}
          onTouchCancel={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <div
            className="button hideOutline block accessibilityOutline tapTransition lg white enabled childrenDiv"
          >
            <div
              className="Text fontSize300 defaultText alignCenter fontWeightSemiBold"
            >
              Visit Pinterest
            </div>
          </div>
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Callout /> message + title 1`] = `
<div
  className="box infoWeak paddingX6 paddingY6 relative rounding4 smDirectionRow smPaddingX8 smPaddingY8 xsDirectionColumn xsDisplayFlex"
>
  <div
    className="box flexWrap smDisplayFlex smMarginBottomN3 smMarginTopN3"
    style={
      Object {
        "width": "100%",
      }
    }
  >
    <div
      className="box justifyCenter smDirectionRow smPaddingY3 xsDirectionColumn xsDisplayFlex xsItemsCenter"
    >
      <div
        className="box marginBottom4 marginTop0 smMarginBottomAuto smMarginTopAuto"
      >
        <svg
          aria-hidden={null}
          aria-label="info"
          className="icon infoIcon iconBlock"
          height={32}
          role="img"
          viewBox="0 0 24 24"
          width={32}
        >
          <path
            d="test-file-stub"
          />
        </svg>
      </div>
      <div
        className="box marginBottomAuto marginTopAuto paddingX6"
        style={
          Object {
            "maxWidth": 648,
          }
        }
      >
        <div
          className="box marginBottomAuto marginTopAuto smDisplayBlock xsDirectionColumn xsDisplayFlex xsItemsCenter"
        >
          <div
            className="box marginBottom2"
          >
            <div
              className="Text fontSize400 defaultText alignStart breakWord fontWeightSemiBold"
            >
              A Title
            </div>
          </div>
          <div
            className="Text fontSize300 defaultText alignStart breakWord fontWeightNormal"
          >
            Insert a clever info callout message here
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
